alternatives: Do not access /var/lib/dpkg/alternatives directly

bdrung opened the pull request:
salt contains modules, which directly access the dpkg internal database, instead of using one of the public interfaces provided by dpkg. This is a problem for several reasons, because even though the layout and format of the dpkg database is administrator friendly, and it is expected that those might need to mess with it, in case of emergency, this “interface” does not extend to other programs besides the dpkg suite of tools. The admindir can also be configured differently at dpkg build or run-time. And finally, the contents and its format, will be changing in the near future.

So use something like:

```
update-alternatives --query $name
```

to query information about alternatives in a machine readable format.

Also add tests cases for `alternatives.show_links` using real data from the distributions.

This is the second merge request to completely address #52605.

Bug: https://github.com/saltstack/salt/issues/52605
Bug-Debian: https://bugs.debian.org/944970

Ch3LL reviewed (changes requested):
This will need a changelog entry.

bdrung committed:
alternatives: Do not access /var/lib/dpkg/alternatives directly

salt contains modules, which directly access the dpkg internal database,
instead of using one of the public interfaces provided by dpkg. This is
a problem for several reasons, because even though the layout and format
of the dpkg database is administrator friendly, and it is expected that
those might need to mess with it, in case of emergency, this “interface”
does not extend to other programs besides the dpkg suite of tools. The
admindir can also be configured differently at dpkg build or run-time.
And finally, the contents and its format, will be changing in the near
future.

So use something like:

```
update-alternatives --query $name
```

to query information about alternatives in a machine readable format.

Also add tests cases for `alternatives.show_links` using real data from
the distributions.

Bug: https://github.com/saltstack/salt/issues/52605
Bug-Debian: https://bugs.debian.org/944970
Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>
